src/pudl/transform/eia.py

```diff
@@ -1157,24 +1149,31 @@ def fix_balancing_authority_codes_with_state(
     return plants
 
 
-eia_assets = load_assets_from_modules([eia860, eia923])
-
-final_eia_table_names = (
-    Package.get_etl_group_tables("entity_eia")
-    + Package.get_etl_group_tables("eia860")
-    + Package.get_etl_group_tables("eia923")
-)
-
-
 @multi_asset(
     ins={
-        asset_key.to_python_identifier(): AssetIn()
-        for eia_asset in eia_assets
-        for asset_key in eia_asset.keys
+        asset_id: AssetIn()
+        for asset_id in [
+            "clean_boiler_fuel_eia923",
+            "clean_boiler_generator_assn_eia860",
+            "clean_boilers_eia860",
+            "clean_coalmine_eia923",
+            "clean_fuel_receipts_costs_eia923",
+            "clean_generation_eia923",
+            "clean_generation_fuel_eia923",
+            "clean_generation_fuel_nuclear_eia923",
+            "clean_generators_eia860",
+            "clean_ownership_eia860",
+            "clean_plants_eia860",
+            "clean_utilities_eia860",
+        ]
```

Comment:
Why did you decide to list the tables out instead of dynamically loading them using:

```python
eia_assets = load_assets_from_modules([eia860, eia923])
```

Reply:
Loading the assets from those modules in this module, and then loading the assets defined in this module at the top level meant that all of the `eia_transform`, `eia860`, and `eia923` modules had to part of the same asset group, which obscured both the clean vs. raw distinction in the `eia860` and `eia923` data sources (raw existed, but clean never did) and also combined the `eia860` and `eia923` asset groups.

I tried to figure out a way to construct the list of assets in this module based on the asset group definitions in `pudl.etl` but couldn't get it to work due to circular imports.

We can simplify the DAG and make the true dependencies clearer within the `eia_transform` group by using the `clean_eia860` and `clean_eia923` groups as inputs to `eia_transform` but restricting its outputs to the entity & annual tables. I think the non-entity table assets in `norm_eia923` and `norm_eia860` should only depend only on their `clean_*` siblings upstream (since all that happens to them in the harvesting step is columns that don't belong in them in the database get dropped, after their useful information has been used to compile the entity & annual tables).

Reply:
Ohh I think I ran into this issue in the [devtool notebook PR](https://github.com/catalyst-cooperative/pudl/pull/2356/files). My workaround was to call `load_assets_from_modules` in the multi_asset decorator:

```python
@multi_asset(
    ins={
        asset_key.to_python_identifier(): AssetIn()
        for eia_asset in load_assets_from_modules([eia860, eia923])
        for asset_key in eia_asset.keys
    },
   ...
)
```
This way the eia860 and eia923 clean assets can be in their own group and you don't have to enumerate all the table names in the decorator. 

Good point about the nonentity tables only depending on the upstream clean tables. I've been meaning to create an issue for disentangling the eia_transform assets to reflect the real dependencies and maybe get some more parallelism going.

Reply:
Are you saying that if I `load_assets_from_modules()` in the decorator (i.e NOT at the top level of the module), then they don't end up having their asset group defined by the top level `load_assets_from_modules(["eia_transform"])` and could be loaded independently with their own asset groups in `pudl.etl`?

I created an issue for the EIA Transform cleanup #2387 that's part of a more general `pudl.etl` DAG cleanup #2386.

Reply:
Yes, that is correct. The assets returned by `load_assets_from_modules` in the decorator aren't included in the module's attributes.

Oh yay thank you for creating those issues!

Reply:
Unfortunately this more elegant and dynamic solution for defining the upstream dependencies results in a (totally unrelated) circular import error, related to the use of `occurrence_consistency() in `pudl.output.eia860` so we need to do some untangling. 🤦🏼  I'm going to punt for the moment and use the hard-coded list of upstream assets.

```
AttributeError: partially initialized module 'pudl' has no attribute 'transform' (most likely due to a circular import)
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/server.py", line 267, in __init__
    self._loaded_repositories: Optional[LoadedRepositories] = LoadedRepositories(
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/server.py", line 116, in __init__
    loadable_targets = get_loadable_targets(
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/utils.py", line 47, in get_loadable_targets
    else loadable_targets_from_python_module(module_name, working_directory)
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_core/workspace/autodiscovery.py", line 36, in loadable_targets_from_python_module
    module = load_python_module(
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_core/code_pointer.py", line 138, in load_python_module
    return importlib.import_module(module_name)
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 992, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/Users/zane/code/catalyst/pudl/src/pudl/__init__.py", line 5, in <module>
    from . import (  # noqa: F401
  File "/Users/zane/code/catalyst/pudl/src/pudl/analysis/__init__.py", line 8, in <module>
    from . import (  # noqa: F401
  File "/Users/zane/code/catalyst/pudl/src/pudl/analysis/state_demand.py", line 38, in <module>
    import pudl.output.pudltabl
  File "/Users/zane/code/catalyst/pudl/src/pudl/output/__init__.py", line 13, in <module>
    from . import (  # noqa: F401
  File "/Users/zane/code/catalyst/pudl/src/pudl/output/eia860.py", line 8, in <module>
    from pudl.transform.eia import occurrence_consistency
  File "/Users/zane/code/catalyst/pudl/src/pudl/transform/__init__.py", line 62, in <module>
    from . import (  # noqa: F401
  File "/Users/zane/code/catalyst/pudl/src/pudl/transform/eia.py", line 1163, in <module>
    [pudl.transform.eia860, pudl.transform.eia923]
```

Reply:
Ah bummer :/ the hard coded list is fine for now.